src/components/Buildbaron/BuildBaronVariables.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useEffect } from "react";
+import { ApolloError, useQuery } from "@apollo/client";
+import { BuildBaronQuery, BuildBaronQueryVariables } from "gql/generated/types";
+import { GET_BUILD_BARON } from "gql/queries";
+
+interface Props {
+  taskId: string;
+  execution: number;
+  setShowbuildbarontab: React.Dispatch<React.SetStateAction<Boolean>>;
+  setbuildBaronData: React.Dispatch<React.SetStateAction<BuildBaronQuery>>;
+  setbuildBaronError: React.Dispatch<React.SetStateAction<ApolloError>>;
+}
+
+export const BuildBaronVariables: React.FC<Props> = ({
+  taskId,
+  execution,
+  setShowbuildbarontab,
+  setbuildBaronData,
+  setbuildBaronError,
+}) => {
+  const {
+    data: buildBaronData,
+    error: buildBaronError,
+    loading: buildBaronLoading,
+    refetch,
+  } = useQuery<BuildBaronQuery, BuildBaronQueryVariables>(GET_BUILD_BARON, {
+    variables: { taskId, execution },
+  });
+
+  useEffect(() => {
```

Comment:
after testing this locally, I see that the useQuery on line 26 will create request new data when the taskId and execution variables on line 27 change. If we convert this function component to a hook, we will actually have to replace `useQuery` with `useLazyQuery` and call refetch with the taskId & execution number from props when taskFailed is true and execution is defined.